resources/git-rebase.md

```diff
@@ -0,0 +1,66 @@
+
+Alas, dealing with merge conflicts is a part of developer life. But they don’t have to be awful! To mitigate conflicts, commit your work often and rebase master when new code is added. If everyone on your team does this, you’ll keep your local branches up-to-date and lower the entire team's odds of having to fix a large merge conflict.
+
+Using the interactive mode when performing a rebase helps as well. Interactive mode can be used by adding the -i flag to your rebase command like so:
+
+ `git rebase -i <nameOfBranchToRebaseIn>`
+
+Using interactive mode when rebasing lets you see the commits that you're rebasing, and allows you to edit and squash commits.
+
+## What is git squash?
+Git squash allows you to take several commits and merge them into a single commit. Say you've finished a feature - styling a nav bar - and your git history looks like this:
+```
+3f0e9cdcda init commit
+9845111670 in progress
+cd362d2f52 still working
+300571e57c styled nav bar
+```
+
+To keep your git history clean and easy to read, you can merge all of these commits into a single commit and update your commit message. You can do this by rebasing your current branch into itself, and then squashing your commits together, like this:
+
+`git rebase -i <nameOfBranchYouAreOn>~4`
+
+The reason "4" is added to the end of the rebase message is because your branch currently has 4 commits that you want to make into a single commit. After you run this command, git will show you the commits you've selected:
+
+```
+pick 3f0e9cdcda init commit
+pick 9845111670 in progress
+pick cd362d2f52 still working
+pick 300571e57c styled nav bar
+```
+
+You can squash your commits by keeping the first commit as is, and selecting 'squash' for the other commits like so:
+
+```
+pick 3f0e9cdcda init commit
+s 9845111670 in progress
+s cd362d2f52 still working
+s 300571e57c styled nav bar
+```
+
+Save your choices, and once you've done this, git will give you the option of editing the commit messages for all of the commits you've changed. You can edit your commit messages into a single commit that describes the work you've done on the branch. When you're finished editing the messages, save and quit your editor. You did it!
+
+## When shouldn't I rebase?
+Once you've pushed your code to your team's repo, don't rebase anymore. If your code is in the repo, other people on your team may have pulled your work and rebased their branches. Rebasing at this point can cause messiness and merge conflicts.
+
+Similairly, once you've put up a PR and someone on your team has reviewed it and requested changes, don't rebase after you make these changes. You want to keep the history of your commits intact at this point so your reviewers can easily follow the work you've updated.
+
+
+If you are curious about what’s going on under the hood when you run `git rebase`, you can read more about the mechanics of it here: https://git-scm.com/book/en/v2/Git-Branching-Rebasing
```

Comment:
Also suggest including this link on the difference between merge and rebase https://www.atlassian.com/git/tutorials/merging-vs-rebasing . :)

Reply:
added!